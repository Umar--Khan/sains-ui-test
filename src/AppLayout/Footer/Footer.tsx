import styled from 'styled-components';
import * as colors from '../../styles/_colors.module.scss';

const StyledFooter = styled.footer`
  background-color: ${colors.greyDark};
  min-height: 6rem;
  color: ${colors.white};
  font-size: 0.9rem;
  font-weight: 400;
`;

const Container = styled.div`
  width: 70%;
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  padding-top: 1rem;
`;

const StyledSpan = styled.span`
  font-weight: 700;
`;

const Footer = () => (
  <StyledFooter>
    <Container>
      <StyledSpan>Sainsbury&rsquo;s, Live Well For Less</StyledSpan>
      <div>&copy; J Sainsbury plc {new Date().getFullYear()}</div>
    </Container>
  </StyledFooter>
);

export default Footer;
