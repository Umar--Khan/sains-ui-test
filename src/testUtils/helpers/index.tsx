import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import { BasketContext, BasketWrapper } from 'src/context/BasketContext';

export const withRouter = (children: any) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export const renderWithBasketContext = (value: BasketWrapper, children: any) =>
  render(
    <BasketContext.Provider value={value}>
      {withRouter(children)}
    </BasketContext.Provider>,
  );
