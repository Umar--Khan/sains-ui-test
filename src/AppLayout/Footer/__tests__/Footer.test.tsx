import renderer from 'react-test-renderer';
import { withRouter } from 'src/testUtils/helpers';
import Footer from '../Footer';

describe('<Footer />', () => {
  it('Should render correctly', () => {
    const component = renderer.create(withRouter(<Footer />));
    expect(component.toJSON()).toMatchSnapshot();
  });
});
