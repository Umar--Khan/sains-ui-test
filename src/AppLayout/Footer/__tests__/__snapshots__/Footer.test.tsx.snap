// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Should render correctly 1`] = `
<footer
  className="sc-bdfBwQ gkkjkl"
>
  <div
    className="sc-gsTCUz efXcWU"
  >
    <span
      className="sc-dlfnbm fXvGP"
    >
      Sainsbury’s, Live Well For Less
    </span>
    <div>
      © J Sainsbury plc 
      2021
    </div>
  </div>
</footer>
`;
